clang/lib/CIR/CodeGen/CIRGenExprScalar.cpp

```diff
@@ -1320,7 +1322,7 @@ mlir::Value ScalarExprEmitter::VisitCastExpr(CastExpr *ce) {
                                      "fixed point casts");
       return {};
     }
-    cgf.getCIRGenModule().errorNYI(subExpr->getSourceRange(), "fp options");
+    assert(!cir::MissingFeatures::cgFPOptionsRAII());
```

Comment:
nit: unrelated change.  Feel free to submit this as review-after-commit.

Reply:
It only looks unrelated. The test case goes through this code path and with the `errorNYI` call here the test returned an error code and so failed.